lib/spack/spack/cmd/common/arguments.py

```diff
@@ -375,6 +375,17 @@ def no_install_status():
     )
 
 
+@arg
+def debug_nondefaults():
+    return Args(
+        "-d",
+        "--debug-nondefaults",
```

Comment:
```suggestion
        "--nondefaults",
```

Reply:
I agree that `--debug--nondefaults` feels a little obtuse. I would also nominate `--highlight-nondefaults` although that is even longer...